### ISSUE TITLE
Reroute node compatibility with Save Image w/ Metadata node (Comfy Image Saver suite)

### ISSUE THREAD

**Issue by alessandroperilli** [rgthree/rgthree-comfy#29]
Partially connected to #24 (the quest to transmit the checkpoint name over long distances), I'm encountering challenges in connecting a Checkpoint Selector node, from Comfy Image Saver suite, to their Save Image w/Metadata node, passing through your reroute node. 

I opened an Issue with them: 

[https://github.com/giriss/comfy-image-saver/issues/12](https://github.com/giriss/comfy-image-saver/issues/12)

This is for visibility and documention only. I'll close it once it gets solved on the other side.

**Comment by rgthree** [comment]
This is now fixed by reusing an existing type-checker in LiteGraph.

However, I should warn you as a workflow distributor, that the types here can get messy.

The reason it wasn't working for your workflow before is my reroute node was checking for strict type compatibility. But, for these list types, they're just the list options (so, in this case, all the checkpoint names you have). But, it looks like nodes lock these types in when they're created. So, your `Checkpoint Selector` node was created at one point in time. Then it looks like you added a new checkpoint and then created the Save Image node.

So, since these were created at two different times, with two different checkpoint lists, the reroute node was seeing these as incompatible types and disconnecting. This is fixed now, and using the same logic in LightGraph itself that allows them to connect in the first place.

However... why this is important for you is your list may look different than someone using your distributed workflow. For the most part, it looks like your workflow will work as-is on any other system because those types are locked. So, just because I have a different set of model filenames and paths doesn't mean I can't use your workflow, but it does mean if I then build on top of your workflow, and of those `ckpt_names` won't connect.

In other words, if I create a new `Checkpoint Selector` node in your workflow, I cannot connect it to any other node already in the workflow that one would assume could connect; not to the existing `Save Image w/Metadata` nodes' `modelname` input, nor the existing `Checkpoint Loader w/ Name (WLSH)` node's `ckpt_name` input, etc. I would have to re-create these nodes from my system to use them.

It's really too bad this is how these input/output types were built. Seems like a bug, but it's rather intentional in the LiteGraph/Comfy code.

**Comment by alessandroperilli** [comment]
I've been called in many ways in my career, but workflow distributor is a first :) 

More seriously, thank you so much for making the changes. They work perfectly and now I have no more flying cables. 
The only place where I still need an auxiliary node, is in the portion below, where I have to convert the list into a string (something I used to do with the `Checkpoint Loader w/ Name (WLSH)` node and now I do with a new node created by on my request: `SD Type Converter`.

<img width="870" alt="Screenshot 2023-10-01 at 11 48 26" src="https://github.com/rgthree/rgthree-comfy/assets/9775949/ac2179dc-7093-4b39-ae30-19312bf43aa8">

It's not ideal, and I hope people won't get confused, but it's the best idea I could come up with. 


Also, thank you so much for the explanation. This makes so much sense now, and it's worrying because it will cause trouble to me and others in the future.

I wish there was a different way to have an isolated `Checkpoint Selector` node, but the one I'm using is the only one I could find across suites. 

I'm also not married to the idea of having individual nodes for the different parameters, but I often found that composite nodes that aggregate them all lack in one way or another. For example, the seed part of composite parameters nodes don't have the great features of your `Seed` node. Or the they don't have the great features of the `CR SDXL Aspect Ratio` node.

In the end, going for a composite parameter node would become too much of a compromise and so I prefer to have individual nodes and "fight" with this `Checkpoint Selector` node. 

On top of fixing these issues, there is a whole range of additional nodes that I'd like to build, but it's not my job and learning how to do this new job would be too much of a distraction at this point.

Thanks for all the help!
